src/BizHawk.Client.EmuHawk/tools/TAStudio/TAStudio.ListView.cs

```diff
@@ -773,29 +763,129 @@ private void TasView_MouseDown(object sender, MouseEventArgs e)
 			}
 		}
 
+		/// <summary>
+		/// Begins a batch of edits, for auto-restore purposes. Auto-restore will be delayed until EndBatchEdit is called.
+		/// </summary>
+		private void BeginBatchEdit()
+		{
+			_batchEditing = true;
+		}
+		/// <returns>Returns true if the input list was redrawn.</returns>
+		private bool EndBatchEdit()
+		{
+			_batchEditing = false;
+			if (_batchEditMinFrame != -1)
+			{
+				return FrameEdited(_batchEditMinFrame);
+			}
+
+			return false;
+		}
+
+		public void ApiHawkBatchEdit(Action action)
+		{
+			// This is only caled from Lua.
+			_editIsFromLua = true;
+			BeginBatchEdit();
+			action();

```

Comment:
...and `try`/`finally` here

Reply:
The Lua call shouldn't be able to throw, but I suppose this could be good if custom tools use it?